requires.py

```diff
@@ -19,22 +18,30 @@
 class SparkRequires(RelationBase):
     scope = scopes.GLOBAL
 
-    def installed(self):
-        return self.get_remote('installed', 'false').lower() == 'true'
+    def is_spark_started(self):
+        return self.get_remote('spark_started', 'false').lower() == 'true'
 
     @hook('{requires:spark}-relation-joined')
     def joined(self):
         conv = self.conversation()
-        conv.set_state('{relation_name}.related')
+        conv.set_state('{relation_name}.joined')
 
     @hook('{requires:spark}-relation-changed')
     def changed(self):
         conv = self.conversation()
-        if self.installed():
-            conv.set_state('{relation_name}.available')
+        if self.is_spark_started():
+            conv.set_state('{relation_name}.ready')
 
     @hook('{provides:spark}-relation-departed')
     def departed(self):
         conv = self.conversation()
-        conv.remove_state('{relation_name}.related')
-        conv.remove_state('{relation_name}.available')
+        conv.remove_state('{relation_name}.joined')
+        conv.remove_state('{relation_name}.ready')
+
+    def get_private_ip(self):
+        conv = self.conversation()
+        return conv.get_remote('private-address')
```

Comment:
Minor nit, the name implies that it will always return an IP address, which is generally not what `private-address` is.  Really, we need to get the DNS charm worked into our deployments and ensure we always return resolvable host names.  Anyway, not really a blocker.